### PR TITLE
Generation of Payement-Reference-Number

### PR DESCRIPTION
**Pull request by moonsglw**

### CODE REVIEW

File: pre-registration-ui/src/app/app.constants.ts [hunk 1]
```diff
@@ -203,4 +203,5 @@ export const BOOKING_DETAIL = {
   appointment_date: '2024-10-01',
   time_slot_from: '09:30:00',
   time_slot_to: '09:45:00'
-};
\ No newline at end of file
+};
+export const generatePRNURL = 'https://apiinternal-mosiphq.nira.go.ug/v1/payment/generatePrn';
```
Comment (Nagarjunakuchi88) [inline pre-registration-ui/src/app/app.constants.ts]: Please move this to pre-reg configuration file.

File: pre-registration-ui/src/app/core/services/data-storage.service.ts [hunk 2]
```diff
@@ -713,11 +719,29 @@ export class DataStorageService {
...
+    const localUrl=appConstants.generatePRNURL;
+    console.log(localUrl);
+    //const headers = new HttpHeaders({ Authorization: `Bearer ${token}`, 'Content-Type': 'application/json',});
+    /*let url =
+      this.BASE_URL +
+      appConstants.APPEND_URL.prn*/
+      return this.httpClient.post<PRNResponse>(localUrl,request);
+    }
+
+    calculateAge(dateStr: string) {
```
Comment (Nagarjunakuchi88) [inline pre-registration-ui/src/app/core/services/data-storage.service.ts:735]: Please let us know use case for this function

File: pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts [hunk 3]
```diff
@@ -701,6 +707,119 @@ export class AcknowledgementComponent implements OnInit, OnDestroy {
     });
   }
 
+
```
Comment (Nagarjunakuchi88) [inline pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts]: please make sure that labels and messages are reading from i18.

File: pre-registration-ui/src/app/app.constants.ts (hunk 1 above)
Comment (rukumbyana) [reply pre-registration-ui/src/app/app.constants.ts]: this was moved

File: pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts (hunk 3 above)
Comment (rukumbyana) [reply pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts]: this was done and the the artfact updated.

File: pre-registration-ui/src/app/core/services/data-storage.service.ts (hunk 2 above)
Comment (rukumbyana) [reply pre-registration-ui/src/app/core/services/data-storage.service.ts:735]: this method is now described in the comment section on top of the method.

File: pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts [hunk 4]
```diff
@@ -701,6 +707,124 @@ export class AcknowledgementComponent implements OnInit, OnDestroy {
...
+   * @description This method checks if the selected desiredService is a payable service 
+   * and checks if age is morethan 16 then calls the generate prn endpoint
+   *it updates the PRN property that is finally rendered on the acknowledgement slip
+   * @private
+   * @memberof AcknowledgementComponent
+   */
+  generatePaymentRefNum(demographicData: any) {
+    const surname = demographicData.surname[0].value;
+    const nin = demographicData.NIN;
+    const desiredService = demographicData.userCase; //userCase may have changed to UserService in qa env
```
Comment (malaypandey-technoforte) [inline pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts]: userCase should be changed to userService as it is renamed in both dev and qa environments

**Review by malaypandey-technoforte — COMMENTED** [review]
need to update acknowledgement.component.ts
const desiredService = demographicData.userCase; 
userCase should be replaced with userService

File: pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts (hunk 4 above)
Comment (rukumbyana) [reply pre-registration-ui/src/app/feature/summary/acknowledgement/acknowledgement.component.ts]: done. changed from userCase to userService

**Review by malaypandey-technoforte — APPROVED** [review]
LGTM

**Pull request merged**